### PR TITLE
chore(deps): add eslint dependencies

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

- Adds missing eslint plugins we're using
- Aligns and updates some test modules so that we're using the same versions in all packages
- Adds a separate `tsconfig.eslint.json` for `typescript-eslint` to include test and stories in linting
- Runs `lint --fix` on all files.

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [ ] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [ ] Doesn't contain any sensitive information

### CODE REVIEW

**Review by burakukula — APPROVED** [review]
Looks good 👍

File: packages/forma-36-react-components/src/components/ControlledInput/README.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { ControlledInput } from './ControlledInput';
+
+# ControlledInput
+
+The ControlledInput component is a styled form component that represents boolean state.
```
Comment (burakukula) [inline packages/forma-36-react-components/src/components/ControlledInput/README.mdx:5]: I think we need to add some more context over here, but I recommend rising separate PR for this purpose. WDYT?

Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/ControlledInput/README.mdx:5]: Woops, this file wasn't supposed to go in this PR 😅

Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/ControlledInput/README.mdx]: Fixed. Good catch, thanks!

**Pull request merged**